\documentclass[ucs,10pt]{beamer}
 
\include{stce-beamer-template}  

\begin{document}
\title[{\tt info@stce.rwth-aachen.de}]{\textcolor{rwth-blue}{Software Lab Computational Engineering Science} \vspace{.2cm} \\ {\small Group 12, Exception Handling}}
\author[Group 12)]{Aaron Floerke, Arseniy Kholod, Xinyang Song and Yanliang Zhu} 
\institute[Software Lab CES]{
{Informatik 12: Software and Tools for Computational Engineering (STCE)} \\ RWTH Aachen University \vspace{.5cm}
}
\date[]{24.06.2024}

\begin{frame}[plain]
\titlepage
\end{frame}

\begin{frame}
	\frametitle{Contents}
\tableofcontents
\end{frame}

\section{Preface}

\begin{frame}
\frametitle{Preface \\
	\small \color{rwth-blue} Exception Handling}

	\begin{itemize}
		\item Software always has a working domain.
		\item User of the software is not aware of all limitations.
		\item Software developer helps user by introducing appropriate exception handling.
		\item Our task is to introduce an exception handling to cppNum v2.4 and v2.5.
	\end{itemize}
\end{frame}

\section{Analysis}

\subsection{User Requirements}

\begin{frame}
\frametitle{Analysis \\
	\small \color{rwth-blue} User Requirements}
	\begin{itemize}
		\item Extend cppNum v2.4 and v2.5 with appropriate C++ exception handling.
		\item Desing at least three scalable sufficiently distinct case studies.
		\item Compare general behavior and run times with the exception handling-free version.
	\end{itemize}
\end{frame}

\subsection{System Requirements}

\begin{frame}
\frametitle{Analysis \\
	\small \color{rwth-blue} System Requirements}
	Functional:
	\begin{itemize}
		\item An exception is thrown, if system is not able to produce correct result, because input data are incorrect or outside the domain.
		\item An exception is thrown before potential crash of the system.
		\item An exception contains an explanatory string.
	\end{itemize}
	Nonfunctional:
	\begin{itemize}
		\item An exception is a class object.
		\item All cppNum exception classes have a single parent class to provide clear structure.
		\item All exception classes are inherited from std::exception to catch together with other exceptions, potentially generated by thirdparty libraries.
	\end{itemize}
\end{frame}

\section{Design}

\subsection{System Requirements}

\begin{frame}
\frametitle{Design \\
	\small \color{rwth-blue} Principal Components and Third-Party Software}
	\begin{itemize}
		\item std::exception to enherit exception classes from.
		\item Eigen library to prove applicability of LU and LLT decompositions.
		\item throw and try-catch mechanism to throw and catch exceptions.
	\end{itemize}
\end{frame}

\subsection{Class Model(s)}

\begin{frame}
\frametitle{Design \\
	\small \color{rwth-blue} Class Model(s)}
\end{frame}

\section{Implementation}

\subsection{Development Infrastructure}

\begin{frame}
\frametitle{Implementation \\
	\small \color{rwth-blue} Development Infrastructure}
\end{frame}

\subsection{Source Code}

\begin{frame}
\frametitle{Implementation \\
	\small \color{rwth-blue} Source Code}
\end{frame}

\subsection{Software Tests}

\begin{frame}
\frametitle{Implementation \\
	\small \color{rwth-blue} Software Tests}
\end{frame}

\section{Project Management}

\begin{frame}
\frametitle{Project Management}
\end{frame}

\section{Live Software Demo}

\begin{frame}
\frametitle{Live Software Demo} 
\end{frame}

\section{Summary and Conclusion}

\begin{frame}
\frametitle{Summary and Conclusion}
\end{frame}

\end{document}
